core/server/api/app.js

```diff
@@ -180,7 +189,14 @@ function apiRoutes() {
     apiRouter.put('/authentication/setup', authenticatePrivate, api.http(api.authentication.updateSetup));
     apiRouter.get('/authentication/setup', api.http(api.authentication.isSetup));
     apiRouter.post('/authentication/token',
-        spamPrevention.signin,
+        brute.globalBlock.prevent,
+        brute.userLogin.getMiddleware({
+            key: function (req, res, next) {
+                req.authInfo = {};
+                req.authInfo.ip = req.ip;
```

Comment:
Two little suggestions here:
`req.ip` returns only the remote client address when trust proxy is disabled. As far as i can remember we have this option enabled and it can happen that req.ip is always the ip from the upstream. Best practise to get the remote client address is using: `req.headers['x-forwarded-for'] ||
  req.connection.remoteAddress`. Also the old implementation of spam-prevention has used `req.connection.remoteAddress`.

Regarding `req.authInfo`. Oh i can totally understand why you are doing that here. I really don't like not having access to the request when `oauth2orize` triggers the exchange callback 😵 How about adding a comment here, because this might be confusing.

```
// oauth2orize supports attaching auth information to be able to access them later (whatever you would like to describe here
if (!req.authInfo) req.authInfo = {}
req.authInfo.ip = req.....
```